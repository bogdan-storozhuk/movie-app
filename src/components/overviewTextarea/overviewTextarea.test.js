import React from "react";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import OverviewTextarea from "./";

describe("OverviewTextarea", () => {
  const props = {
    handleChange: jest.fn(),
    handleBlur: jest.fn(),
    name: "overview",
    value: "",
  };
  const filledValue = "test";
  it("should match snapshot with empty value", () => {
    const { asFragment } = render(<OverviewTextarea {...props} />);

    expect(asFragment(<OverviewTextarea {...props} />)).toMatchSnapshot();
  });

  it("should match snapshot with empty value", () => {
    const { asFragment } = render(
      <OverviewTextarea {...props} value={filledValue} />
    );

    expect(
      asFragment(<OverviewTextarea {...props} value={filledValue} />)
    ).toMatchSnapshot();
  });

  it("should correctly handle handleChange event", () => {
    render(<OverviewTextarea {...props} value="" />);
    const textarea = screen.queryByDisplayValue("");
    expect(textarea).toBeInTheDocument();
    userEvent.type(textarea, filledValue);
    expect(props.handleChange).toHaveBeenCalled();
  });
});
