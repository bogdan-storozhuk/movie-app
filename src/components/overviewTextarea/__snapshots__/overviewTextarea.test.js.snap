// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewTextarea should match snapshot with empty value 1`] = `
<DocumentFragment>
  <textarea
    class="OverviewTextarea"
    id="overview"
    name="overview"
    type="text"
  />
</DocumentFragment>
`;

exports[`OverviewTextarea should match snapshot with empty value 2`] = `
<DocumentFragment>
  <textarea
    class="OverviewTextarea"
    id="overview"
    name="overview"
    type="text"
  >
    test
  </textarea>
</DocumentFragment>
`;
